tests/codegen/zst-offset.rs

```diff
@@ -34,7 +34,7 @@ pub struct U64x4(u64, u64, u64, u64);
 // CHECK-LABEL: @vector_layout
 #[no_mangle]
 pub fn vector_layout(s: &(U64x4, ())) {
-// CHECK: getelementptr i8, {{.+}}, [[USIZE]] 32
+// CHECK: getelementptr inbounds i8, {{.+}}, [[USIZE]] 32
```

Comment:
As mentioned above, these ZSTs are one past the end, so they should be fine. Not sure if there are untested cases where they can be further than that.